app/views/catalog/code_snippet.html.erb

```diff
@@ -0,0 +1 @@
+<%= render Earthworks::CodeSnippetModalComponent.new(@docs.first) %>
```

Comment:
I think we can simplify this whole logic around `@docs` and the block you've added to the catalog controller by using the `document` that is passed via the sidebar controller ` Earthworks::OtherOptionsComponent.new(document: document)`

Happy to discuss this.

Reply:
We discussed this ~~ food for thought.